### PR TITLE
Disable smooth turning for 180 degree turns.

### LINKED ISSUE #11857: System.DivideByZeroException in the MovePart activity
No idea how I got that. I was just ordering an attack move while scrolling (actually, I think the game had a bug there and let my viewport jump to the bottom left edge, but I can't reproduce that reliably either).

```
Red Alert Mod at Version {DEV_VERSION}
on map 83e3d202a01e95b77fd776562aa418c7dca56a97 (04:  Ten to one by Westwood Studios).
Date: 2016-08-21 16:38:04Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.DivideByZeroException`: Attempted to divide by zero. (Es wurde versucht, durch 0 (null) zu teilen.)
   at OpenRA.Mods.Common.Activities.Move.MovePart..ctor(Move move, WPos from, WPos to, Int32 fromFacing, Int32 toFacing, Int32 startingFraction)
   at OpenRA.Mods.Common.Activities.Move.MoveFirstHalf..ctor(Move move, WPos from, WPos to, Int32 fromFacing, Int32 toFacing, Int32 startingFraction)
   at OpenRA.Mods.Common.Activities.Move.MoveFirstHalf.OnComplete(Actor self, Mobile mobile, Move parent)
   at OpenRA.Mods.Common.Activities.Move.MovePart.InnerTick(Actor self, Mobile mobile)
   at OpenRA.Mods.Common.Activities.Move.MovePart.Tick(Actor self)
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act) in \OpenRA\OpenRA.Game\Traits\ActivityUtils.cs:Line 37.
   at OpenRA.Mods.Common.Activities.AttackMoveActivity.Tick(Actor self)
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act) in \OpenRA\OpenRA.Game\Traits\ActivityUtils.cs:Line 37.
   at OpenRA.Actor.Tick() in \OpenRA\OpenRA.Game\Actor.cs:Line 164.
   at OpenRA.World.Tick() in \OpenRA\OpenRA.Game\World.cs:Line 339.
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in \OpenRA\OpenRA.Game\Game.cs:Line 561.
   at OpenRA.Game.LogicTick() in \OpenRA\OpenRA.Game\Game.cs:Line 586.
   at OpenRA.Game.Loop() in \OpenRA\OpenRA.Game\Game.cs:Line 713.
   at OpenRA.Game.Run() in \OpenRA\OpenRA.Game\Game.cs:Line 753.
   at OpenRA.Program.Run(String[] args) in \OpenRA\OpenRA.Game\Support\Program.cs:Line 119.
   at OpenRA.Program.Main(String[] args) in \OpenRA\OpenRA.Game\Support\Program.cs:Line 40.
```

### PR DESCRIPTION
**Pull request by pchote**
Fixes #11857.

### CODE REVIEW

**Comment by obrakmann** [comment]
:+1:

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/a80fc3eb344a0826f4307e4172a8e9a03da8a6ac...6168e5293edc9fea42ce918a90ae63076c840a8a)

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 1]
```diff
@@ -312,7 +312,8 @@ public MovePart(Move move, WPos from, WPos to, int fromFacing, int toFacing, int
 				MoveFractionTotal = (to - from).Length;
 
 				// Calculate an elliptical arc that joins from and to
-				if (fromFacing != toFacing)
+				var delta = Util.NormalizeFacing(fromFacing - toFacing);
+				if (delta != 0 && delta != 128)
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: This needs to be an || not an &&!

Comment (GraionDilach) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: Oh wait, no. Still, this doesn't catch it.

pillanatfelvetel1

Comment (pchote) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: No, that would make it run unconditionally.  If a number is == 0 then it certainly != 128, and vice versa.

**Pull request merged**